server/py/framework/utils/notifications/notification_pusher.py

```diff
@@ -162,9 +165,37 @@ async def async_push():
                         "Failed to push notification async",
                         error=mlrun.errors.err_to_str(result),
                     )
+            if activation_id and activation_time:
+                try:
+                    self._update_alert_activation_notification_state(
+                        activation_id=activation_id,
+                        activation_time=activation_time,
+                        alert=alert,
+                    )
+                except Exception as e:
+                    logger.warning(
+                        "Failed to update alert activation state", error=str(e)
+                    )
```

Comment:
Add a comment that you do it here to ensure that the notifications are updated with their actual status.

Reply:
@TomerShor it is not really "ensure" -  I always save notification state only here. We don't save it anymore on initial storing of alert activation.

Reply:
Right, so that's what I wanted to be in the comment.
To make it clear that the notifications are only saved here after their status is updated.